sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -225,16 +230,6 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
     }.isDefined
   }
 
-  /**
-   * To be able to prune partitions on a join key, the filtering side needs to
-   * meet the following requirements:
-   *   (1) it can not be a stream
-   *   (2) it needs to contain a selective predicate used for filtering
```

Comment:
IIUC, the eventual goal is to make sure the "filtering side" is small. How do you guarantee it without selective predicates?

Reply:
In your example, the "filtering side" is a very small parquet table with 300+ rows. I don't think that's a general case and can justify removing the selective predicate check.

Reply:
e.g. we may broadcast a table with 1 million rows, and DPP may not be beneficial in that case.

Reply:
Base on another case https://github.com/apache/spark/pull/34070#issuecomment-973893038. May be we can support DPP if filtering side rows less than 1000(`spark.sql.optimizer.dynamicPartitionPruning.filteringSideThreshold`).